awx/api/serializers.py

```diff
@@ -4453,6 +4454,122 @@ def validate(self, attrs):
+    # if we can find out the user, we can filter down the UnifiedJobTemplate objects
+    unified_job_template = serializers.PrimaryKeyRelatedField(queryset=UnifiedJobTemplate.objects.all(), many=False)
+    inventory = serializers.PrimaryKeyRelatedField(queryset=Inventory.objects.all(), required=False, many=False)
+    credentials = serializers.PrimaryKeyRelatedField(queryset=Credential.objects.all(), required=False, many=True)
+    identifier = serializers.CharField(required=False, write_only=True, allow_blank=False)
+
+    class Meta:
+        fields = (
+            'unified_job_template',
+            'identifier',
+            'inventory',
+            'credentials',
+            #   'labels',
+            #   'extra_data',
+            #   'survey_passwords',
+            #   'char_prompts',
+            #   'labels',
+            #   'instance_groups',
+            #   'execution_environment',
+        )
+
+    def validate(self, attrs):
+        # Since we don't validate using the model here, I don't want to pass any extra things that we are not
+        # explicitly checking RBAC on up to the BulkJobsLaunchSerializer because we could allow people to launch with
+        # related items they don't have permission to
+        for key in attrs.keys():
+            if key not in self.fields:
+                attrs.pop(key)
+        attrs = super().validate(attrs)
+        return attrs
+
+
+class BulkJobLaunchSerializer(serializers.Serializer):
+    name = serializers.CharField(max_length=512, required=False)  # limited by max name of jobs
+    jobs = serializers.ListField(child=BulkJobNodeSerializer(), allow_empty=False, max_length=100)
+
+    class Meta:
+        fields = ('name', 'jobs')
+        read_only_fields = ()
+
+    def validate_jobs(self, jobs):
+        view = self.context.get('view', None)
+        identifiers = set()
+        for node in jobs:
+            if 'identifier' in node:
+                if node['identifier'] in identifiers:
+                    raise serializers.ValidationError(_(f"Identifier {identifier} not unique"))
+                identifiers.add(node['identifier'])
+            else:
+                node['identifier'] = str(uuid4())
+        if view and not view.request.user.is_superuser:
+            allowed_ujts = set()
+            allowed_ujts.union(set(JobTemplate.allowed_objects(request.user, 'execute_role').all()))
+            allowed_ujts.union(set(WorkflowJobTemplate.allowed_objects(request.user, 'execute_role').all()))
+            allowed_ujts.union(set(ProjectUpdate.allowed_objects(request.user, 'update_role').all()))
+            accessible_inventories_qs = Inventory.accessible_objects(request.user, 'update_role')
+            allowed_ujts.union(set(InventorySource.objects.filter(inventory__in=accessible_inventories_qs.all())))
+            requested_ujts = set(UnifiedJobTemplate.objects.filter(id__in=[j['unified_job_template'] for j in jobs]))
```

Comment:
doing the query shouldn't be necessary, this line could just be

```
requested_ujts = {j['unified_job_template'] for j in jobs}
```
though that assumes that you build up the `allowed_ujts` as pks instead of objects (which I totally think you should).